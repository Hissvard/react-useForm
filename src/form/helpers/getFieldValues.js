const getListFieldValues = items => {
  return items.reduce((acc, item) => {
    const itemValues = getFieldValues(item)
    acc.push(itemValues)
    return acc
  }, [])
}

export const getFieldValues = (fieldData) => {
  return Object.entries(fieldData).reduce((acc, [k, v]) => {
    let value
    if (v.props) {
      if (v.props.items) {
        value = getListFieldValues(v.props.items)
      } else {
        value = v.props.value
      }
    } else {
      if (v.items) {
        value = getListFieldValues(v.items)
      } else {
        value = v.value
      }
    }

    acc[k] = value
    return acc
  }, {})
}

