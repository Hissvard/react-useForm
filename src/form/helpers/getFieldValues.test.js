import { getFieldValues } from './getFieldValues'

describe('getFieldValues', () => {

  test('gets nested values', () => {
    const fieldData = {
      nickname: {
        props: {
          value: 'nickname',
        },
      },
      fullName: {
        props: {
          value: 'fullName',
        },
      },
      colors: {
        props: {
          items: [
            {
              color: {
                value: 'color1',
              },
              shades: {
                items: [
                  { shade: { value: 'shade-1-1' } },
                ],
              },
            },
            {
              color: {
                value: 'color2',
              },
            },
          ],
        },
      },
    }

    expect(getFieldValues(fieldData)).toMatchSnapshot()
  })

})
