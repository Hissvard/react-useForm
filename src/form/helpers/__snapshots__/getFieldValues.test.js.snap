// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getFieldValues gets nested values 1`] = `
Object {
  "colors": Array [
    Object {
      "color": "color1",
      "shades": Array [
        Object {
          "shade": "shade-1-1",
        },
      ],
    },
    Object {
      "color": "color2",
    },
  ],
  "fullName": "fullName",
  "nickname": "nickname",
}
`;
